jdi-light/src/main/java/com/epam/jdi/light/settings/WebSettings.java

```diff
@@ -115,64 +134,220 @@ private static void fillAction(JAction1<String> action, String name) {
+    }
+
+    private TextTypes getTextType(String type) {
+        TextTypes textType = first(getAllEnumValues(TextTypes.class),
+                t -> t.toString().equals(type));
+        return textType != null
+                ? textType : SMART_TEXT;
+    }
+
+    private SetTextTypes getSetTextType(String type) {
+        SetTextTypes textType = first(getAllEnumValues(SetTextTypes.class),
+                t -> t.toString().equals(type));
+        return textType != null
+                ? textType : SET_TEXT;
+    }
+
+    private String getRemoteUrl(String prop) {
+        switch (prop.toLowerCase().replaceAll(" ", "")) {
+            case "sauce":
+            case "saucelabs":
+                getJDISettings().DRIVER.capabilities.common = sauceCapabilities();
+                return sauceLabs();
+            case "browserstack":
+                return browserstack();
+            default:
+                return seleniumLocalhost();
+        }
+    }
+
+    private Pair<String, JFunc1<String, String>> getSmartSearchFunc(String name) {
+        if (!SMART_MAP_NAME_TO_LOCATOR.keys().contains(name)) {
+            throw exception("Unknown JDISettings.ELEMENT.smartName: '%s'. Please correct value 'smart.locators.toName' in test.properties." +
+                    "Available names: [%s]", name, print(SMART_MAP_NAME_TO_LOCATOR.keys()));
+        }
+        return Pair.$(name, SMART_MAP_NAME_TO_LOCATOR.get(name));
+    }
+
+    private UseSmartSearch getSmartSearchUse(String prop) {
+        String propLower = prop.toLowerCase().trim().replaceAll(" ", "");
+        switch (propLower) {
+            case "false":
+                return FALSE;
+            case "onlyui":
+                return ONLY_UI;
+            case "uiandelements":
+                return UI_AND_ELEMENTS;
+            case "always":
+                return ALWAYS;
+            default:
+                return UI_AND_ELEMENTS;
+        }
+    }
+
+    private boolean getBoolean(String param) {
         String lowerParams = param.toLowerCase();
         return !lowerParams.equals("off") && !lowerParams.equals("false");
     }
 
-    public static void loadCapabilities(String property, String defaultPath, JAction1<Properties> setCapabilities) {
+    public void loadCapabilities(String property, String defaultPath, JAction1<Properties> setCapabilities) {
         String path = "";
         try {
             path = System.getProperty(property, getProperty(property));
-        } catch (Exception ignore) { }
+        } catch (Exception ignore) {
+        }
         if (isEmpty(path))
             path = defaultPath;
         Properties properties = new PropReader(path).getProperties();
         if (properties.isEmpty())
             return;
         try {
             setCapabilities.execute(properties);
-        } catch (Exception ignore) { }
+        } catch (Exception ignore) {
+        }
+    }
+
+    private void setSearchStrategy(String param) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'setSearchStrategy(String)' has an NPath complexity of 580](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5482899)